### PR TITLE
[TIMOB-18337] iOS: Adding building and injecting extensions from external Xcode projects

### PR DESCRIPTION
**Pull request by jonalter**
Testing instructions in ticket [TIMOB-18337](https://jira.appcelerator.org/browse/TIMOB-18337)

### CODE REVIEW

File: apidoc/Titanium/App/iOS/UserDefaults.yml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    The UserDefaults module is used for storing application-related data in property/value pairs 
+    that persist beyond application sessions and device power cycles. UserDefaults allows the suiteName
+    of the UserDefaults to be specified at creation time.
+description: |
+    Unlike Titanium.App.Properties, Titanium.App.iOS.UserDefaults does not pull properties defined in the `tiapp.xml` file.
+
+    To create a notification action, use the <Titanium.App.iOS.createUserDefaults> method.
+creatable: false
+extends: Titanium.Proxy
+since: "4.0.0"
```
Comment (bhatfield) [inline apidoc/Titanium/App/iOS/UserDefaults.yml]: Add `platforms: [iphone, ipad]`

File: apidoc/Titanium/App/iOS/UserDefaults.yml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      - name: default
+        summary: Default value to return if property does not exist.
+        type: Boolean
+        optional: true
+        
+  - name: getDouble
+    summary: Returns the value of a property as a double (double-precision, floating point) data type.
+    description: |
+        This method must only be used to retrieve properties created with 
+        [setDouble](Titanium.App.Properties.setDouble).
```
Comment (bhatfield) [inline apidoc/Titanium/App/iOS/UserDefaults.yml]: Should the links be changed from `Titanium.App.Properties` to `Titanium.App.iOS.UserDefaults`?

Repeat for `getInt` and `setInt` methods.

File: apidoc/Titanium/App/iOS/UserDefaults.yml [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+examples:
+  - title: Create a UserDefaults object
+    example: |
+        Creating a UserDefaults object
+        
+            var props = Ti.App.iOS.createUserDefaults({
+                suiteName: 'group.mySuite'
+            });
+
+  - title: Store a property
```
Comment (bhatfield) [inline apidoc/Titanium/App/iOS/UserDefaults.yml]: Um? Are these two example applicable here?

File: apidoc/Titanium/App/iOS/UserDefaults.yml [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+summary: |
+    The UserDefaults module is used for storing application-related data in property/value pairs 
+    that persist beyond application sessions and device power cycles. UserDefaults allows the suiteName
+    of the UserDefaults to be specified at creation time.
+description: |
+    Unlike Titanium.App.Properties, Titanium.App.iOS.UserDefaults does not pull properties defined in the `tiapp.xml` file.
+
+    To create a notification action, use the <Titanium.App.iOS.createUserDefaults> method.
+creatable: false
+extends: Titanium.Proxy
```
Comment (bhatfield) [inline apidoc/Titanium/App/iOS/UserDefaults.yml]: I think you could have extended from Titanium.App.Properties, so you didn't have to repeat the docs.

File: apidoc/Titanium/Filesystem/Filesystem.yml [hunk 5]
```diff
@@ -80,6 +92,23 @@ methods:
...
         type: Titanium.Filesystem.FileStream
     exclude-platforms: [blackberry]
+  - name: directoryForSuite
+    summary: Returns the path to the container directory associated with the specified security application group ID.
+    description: |
+        This is a writable directory used on iOS. For groups of sandboxed apps that need to share files 
+        and other information can request a container directory as part of their entitlements.
+        
+        When called with a valid group identifier, this method returns the location of that directory as a string.
+        Returns null if the group identifier is invalid, check the app's entitlements.
```
Comment (bhatfield) [inline apidoc/Titanium/Filesystem/Filesystem.yml]: change comma to semicolon.

File: apidoc/Titanium/Filesystem/Filesystem.yml [hunk 6]
```diff
@@ -80,6 +92,23 @@ methods:
     returns:
         type: Titanium.Filesystem.FileStream
     exclude-platforms: [blackberry]
+  - name: directoryForSuite
+    summary: Returns the path to the container directory associated with the specified security application group ID.
+    description: |
+        This is a writable directory used on iOS. For groups of sandboxed apps that need to share files
```
Comment (bhatfield) [inline apidoc/Titanium/Filesystem/Filesystem.yml]: Remove `For` and capitalize `groups`

**Comment by pec1985** [comment]
Code reviewed and tested.

**Pull request merged**